subiquity/common/filesystem/tests/test_gaps.py

```diff
@@ -165,3 +167,153 @@ def test_half_extended_half_logical(self):
                 gaps.Gap(d, 35, 15, True),
                 gaps.Gap(d, 50, 50, False),
             ])
+
+
+class TestMovableTrailingPartitionsAndGapSize(unittest.TestCase):
+
+    def use_alignment_data(self, alignment_data):
+        m = mock.patch('subiquity.common.filesystem.gaps.parts_and_gaps')
+        p = m.start()
+        self.addCleanup(m.stop)
+        p.side_effect = partial(
+            gaps.find_disk_gaps_v2, info=alignment_data)
+
+    def test_no_next_gap(self):
+        self.use_alignment_data(PartitionAlignmentData(
+            part_align=10, min_gap_size=1, min_start_offset=10,
+            min_end_offset=10, primary_part_limit=10))
+        # 0----10---20---30---40---50---60---70---80---90---100
+        # #####[ p1                                   ]#####
+        m, d = make_model_and_disk(size=100)
+        p = make_partition(m, d, offset=10, size=80)
+        self.assertEqual(
+            ([], 0),
+            gaps.movable_trailing_partitions_and_gap_size(p))
+
+    def test_immediately_trailing_gap(self):
+        self.use_alignment_data(PartitionAlignmentData(
+            part_align=10, min_gap_size=1, min_start_offset=10,
+            min_end_offset=10, primary_part_limit=10))
+        # 0----10---20---30---40---50---60---70---80---90---100
+        # #####[ p1      ]         [ p2 ]              #####
```

Comment:
I love the ASCII art diagrams!
Nitpick: there is a slight inconsistency in the placement of the end bracket and the intended end position of the partition in the test, compared to some of the others such as `test_one_trailing_movable_partition_and_gap`.  This nitpick may or may not be related to me squinting at off-by-one errors in the curtin resize tests.

Reply:
> I love the ASCII art diagrams!

Me too! I wondered about implementing a parser for them so they could actually be the input to the test...

> Nitpick: there is a slight inconsistency in the placement of the end bracket and the intended end position of the partition in the test, compared to some of the others such as `test_one_trailing_movable_partition_and_gap`. This nitpick may or may not be related to me squinting at off-by-one errors in the curtin resize tests.

Even more of a reason to delete the test!